charts/argo-workflows/templates/controller/workflow-rb.yaml

```diff
@@ -1,19 +1,18 @@
 {{- if .Values.workflow.rbac.create -}}
+  {{- range $namespace := append .Values.controller.workflowNamespaces .Values.workflow.namespace | compact | uniq }}
```

Comment:
Looks good. Resolved in https://github.com/argoproj/argo-helm/pull/824/commits/094cd02860911da4e31e0c4f62f964d2f89dbe91

Reply:
Can you please also wrap the namespaces into `with` blocks like above? Without that, the template will generate `namespace: ""` when `.Values.singeNamespace` is `true`.

Reply:
Also, I messed up the loop definition. It should be
```
{{- range $namespace := .Values.singeNamespace | ternary (list "") (append .Values.controller.workflowNamespaces .Values.workflow.namespace | compact | uniq) }}
```
otherwise the chart expansion will fail.

Reply:
> Can you please also wrap the namespaces into `with` blocks like above? Without that, the template will generate `namespace: ""` when `.Values.singeNamespace` is `true`.

Oh oops. Fixed in https://github.com/argoproj/argo-helm/pull/824/commits/2bfcf1bc0e9a031dbe5c5ff0c0e9d524da742f75

Reply:
> Also, I messed up the loop definition. It should be
> 
> ```
> {{- range $namespace := .Values.singeNamespace | ternary (list "") (append .Values.controller.workflowNamespaces .Values.workflow.namespace | compact | uniq) }}
> ```
> 
> otherwise the chart expansion will fail.

Yeah I was wondering about the `concat` instead of the `append`, but no worries. Fixed in https://github.com/argoproj/argo-helm/pull/824/commits/ff83f5e3f48f2bdbbce119625d10036b21394fcc